src/app/components/site/Homepage.tsx

```diff
@@ -1,173 +1,155 @@
-                                            {"We also offer free "}
-                                            <Link to="/events?types=teacher">teacher CPD events</Link>{" and "}
-                                            <Link to="/events?types=student">student workshops</Link>.<br />
-                                            {"Isaac Computer Science is proud to be part of the Department for Education's "}
-                                            <Link to="/teachcomputing">National Centre for Computing Education</Link>.
-                                        </p>
-                                    </Col>
-                                </Row>
-                            </Col>
-                            <Col lg="7" className="order-last order-lg-1 px-lg-5 align-self-center text-center pattern-03">
-                                <iframe
-                                    title="Isaac Computer Science introduction video" width="640" height="345"
-                                    className="mw-100 pt-lg-4"
-                                    src="https://www.youtube-nocookie.com/embed/ci6_Du_NHZA?enablejsapi=1&rel=0&fs=1&modestbranding=1&origin=home"
-                                    frameBorder="0" allowFullScreen
-                                />
-                            </Col>
-                            <Col className="order-lg-last pb-5 pb-lg-3">
-                                <ShowMeButtons />
-                            </Col>
-                        </Row>}
-                </Container>
-            </section>
+  const promoItem = promo ? promo[0] : undefined;
 
-            {!user?.loggedIn && <Container>
-                <hr/>
-            </Container>}
+  const featuredNewsItem = news && user?.loggedIn ? news[0] : undefined;
 
-            {!user?.loggedIn && <section id="why-sign-up" className="row sign-up-tabs">
-                <Container>
-                    <Col className="pb-5 pt-4 pattern-04">
-                        <h2 className="text-center mb-5">Why sign up?</h2>
-                        <WhySignUpTabs/>
-                    </Col>
-                </Container>
-            </section>}
+  let carouselNewsItems: IsaacPodDTO[] = [];
 
-            <section id="news">
-                <Container className={classNames("pt-4 pb-5", {"mt-lg-n5 pt-lg-0": user?.loggedIn ?? false})}>
-                    <div data-testid={"news-carousel"} className="eventList pt-5 pattern-03-reverse">
-                        <h2 className="h-title mb-4">News</h2>
-                        {user?.loggedIn && <div className={"d-block d-lg-none mb-4 mb-lg-0"}>
-                            <FeaturedNewsItem item={featuredNewsItem} />
-                        </div>}
-                        <NewsCarousel items={carouselNewsItems} />
-                    </div>
-                </Container>
-            </section>
+  if (news) {
+    (user?.loggedIn && user?.role === "TEACHER" && promoItem) ||
```

Comment:
This is a very unconventional way to use the ternary operator. You normally use the ternary operator when you want to assign the result of either branch to some variable. The way you've written it, this may as well be a normal if/else.